common/main/cpp/native_c4.cc

```diff
@@ -138,6 +139,26 @@ static void logCallback(C4LogDomain domain, C4LogLevel level, const char *fmt, v
     }
 }
 
+/*
+ * We let our code create new domains ("true" as 2nd arg to c4log_getDomain) if it wants.
+ * The advantage is that if, esp for debugging, we want to log to a dynamically created domain,
+ * we can do so at any time, including before Core creates it.  The problem with that is that
+ * LiteCore doesn't keep the names, itself.  Unless the domain already exists, it will use
+ * the string that we pass to it.  That means that string has to be stored somewhere permanent.
+ * It is possible that this can go away when CBL-5726 is resolved
+ */
+namespace litecore {
+    namespace jni {
+        static std::unordered_set<std::string> gDomainNames;
+
+        // Assert: jdomain is not null
```

Comment:
null would be fine, LiteCore auto maps null to empty string aka default

Reply:
I think this will fail on null:
`std::string domain = JstringToUTF8(env, jdomain);`